discretisedfield/tests/test_field.py

```diff
@@ -2201,3 +2200,78 @@ def test_numpy_ufunc(self):
                          value=lambda _: np.random.random(3)*2 - 1)
         assert np.allclose(np.exp(field.orientation).array,
                            np.exp(field.orientation.array))
+
+    def test_to_xarray_valid_args(self):
+        for mesh in self.meshes:
+            for value, dtype in self.vfuncs:
+                f = df.Field(mesh, dim=3, value=value, dtype=dtype)
+                fxa = f.to_xarray()
+                assert f.dim == fxa['comp'].size
+                assert isinstance(fxa, xr.DataArray)
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('x'))),
+                    fxa.x.values
+                )
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('y'))),
+                    fxa.y.values
+                )
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('z'))),
+                    fxa.z.values
+                )
+                assert np.array_equal(f.array, fxa.values)
+                assert all(
+                    fxa[i].attrs['units'] == f.mesh.attributes['unit']
+                    for i in ['x', 'y', 'z']
+                )
+                assert all(fxa['comp'].values == f.components)
+
+            for value, dtype in self.sfuncs:
+                f = df.Field(mesh, dim=1, value=value, dtype=dtype)
+                fxa = f.to_xarray()
+                assert isinstance(fxa, xr.DataArray)
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('x'))),
+                    fxa.x.values
+                )
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('y'))),
+                    fxa.y.values
+                )
+                assert np.array_equal(
+                    np.array(list(f.mesh.axis_points('z'))),
+                    fxa.z.values
+                )
+                assert 'comp' not in fxa.dims
+                assert np.array_equal(f.array.squeeze(axis=-1), fxa.values)
+                assert all(
+                    fxa[i].attrs['units'] == f.mesh.attributes['unit']
+                    for i in ['x', 'y', 'z']
+                )
+
+        f6d = self.pf << self.pf
+        f6d_xa = f6d.to_xarray()
+        assert f6d_xa['comp'].size == 6
```

Comment:
Can we test the default (i.e. not set) `comp` labels?

Reply:
Added 
```Python 
assert 'comp' not in f6d_xa.coords
```